src/Npgsql/NpgsqlConnector.cs

```diff
@@ -1440,11 +1458,20 @@ internal void CancelRequest(bool throwExceptions = false)
                     var socketException = e.InnerException as SocketException;
                     if (socketException == null || socketException.SocketErrorCode != SocketError.ConnectionReset)
                     {
+                        if (requestedByUser)
+                            ReadCts.Timeout = TimeSpan.Zero;
```

Comment:
Can you provide some info on what this does?

Reply:
Yeah, this parts logic isn't as straightforward as I would have liked. I'll take a look tomorrow, and with the write side changes it should become much more easier to follow.